client/app/scripts/superdesk-search/search.js

```diff
@@ -1020,6 +1026,62 @@
                     });
+
+                    metadata.fetchSubjectcodes().then(function () {
+                        scope.subjectcodes = metadata.values.subjectcodes;
+                        tags.initSelectedFacets().then(function (currentTags) {
+                            scope.subjectitems = {
+                                subject: getSubjectCodes(currentTags)
+                            };
+                        });
+                    });
+
+                    function getSubjectCodes(currentTags) {
+                        var queryArray = currentTags.selectedParameters, filteredArray = [];
+                        if (!$location.search().q) {
+                            return filteredArray;
+                        }
+                        for (var i = 0, queryArrayLength = queryArray.length; i < queryArrayLength; i++) {
+                            var queryArrayElement = queryArray[i];
+                            if (queryArrayElement.indexOf('subject.name') !== -1) {
+                                var elementName = queryArrayElement.substring(
+                                        queryArrayElement.lastIndexOf('(') + 1,
+                                        queryArrayElement.lastIndexOf(')')
+                                        );
+                                for (var j = 0, subjectCodesLenght = scope.subjectcodes.length; j < subjectCodesLenght; j++) {
+                                    if (scope.subjectcodes[j].name === elementName) {
+                                        filteredArray.push(scope.subjectcodes[j]);
+                                    }
+                                }
+                            }
+                        }
+                        return filteredArray;
+                    }
+
+                    scope.subjectSearch = function (item) {
+                        tags.initSelectedFacets().then(function (currentTags) {
+                            if (item.subject.length > getSubjectCodes(currentTags).length) {
+                                var addItemSubjectName = 'subject.name:(' + item.subject[item.subject.length - 1].name + ')';
+                                var query = getQuery(),
+                                        q = (query === null ?
+                                                addItemSubjectName :
+                                                query + ' ' + addItemSubjectName);
+
+                                $location.search('q', q);
+                            } else {
+                                var params = $location.search();
+                                if (params.q) {
+                                    for (var j = 0; j < getSubjectCodes(currentTags).length; j++) {
+                                        if (item.subject.indexOf(getSubjectCodes(currentTags)[j]) === -1) {
+                                            var removeItemSubjectName = 'subject.name:(' + getSubjectCodes(currentTags)[j].name + ')';
```

Comment:
`getSubjectCodes(currentTags)` is repeated multiple times, better to assign it to a variable at the beginning of the function, instead of repeatedly invoking the `getSubjectCodes()` function.